### PR TITLE
Upgrade for Node 18 compatibility

### PR DESCRIPTION
**Pull request by javiertoledo**
In this branch we're working on a big upgrade to remove deprecated dependencies and start using Node 18.

## TODO

- [x] Raised the node version to 18.18.0
- [x] Review why the core & cli packages unit tests are failing
- [x] Review why the AWS package unit tests are failing
- [x] Increase the node version for Azure Functions - currently set to node 14

## Checks
- [ ] Project Builds
- [ ] Project passes tests and checks
- [ ] Updated documentation accordingly

### CODE REVIEW

File: packages/framework-provider-aws-infrastructure/src/infrastructure/params.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function lambda(
   apis: APIs
 ): Pick<FunctionProps, 'runtime' | 'timeout' | 'memorySize' | 'environment'> {
   return {
-    runtime: Runtime.NODEJS_14_X,
+    runtime: Runtime.NODEJS_18_X,
```
Comment (gonzalogarciajaubert) [inline packages/framework-provider-aws-infrastructure/src/infrastructure/params.ts:21]: Azure infrastructure also defines the runtime version in **terraform-function-app.ts** file. Current version is Node 14:

```
      siteConfig: {
        applicationStack: {
          nodeVersion: '~14',
        },
      },
```

I think we should update it too

Comment (gonzalogarciajaubert) [reply packages/framework-provider-aws-infrastructure/src/infrastructure/params.ts:21]: Also I think the `functionsExtensionVersion: '~3'` needs to be updated to `functionsExtensionVersion: '~4'` in the same file

**Comment by javiertoledo** [comment]
@samueldominguez very good observation, let's upgrade the SDK to v3 then before merging this branch. I'll add a note in the description as a reminder.

**Comment by javiertoledo** [comment]
I've moved all pending tasks to separate issues and retargeted this PR to the new `v2.x.x` base branch. Both unit and integration tests are passing, so I'll merge it.

**Pull request merged**